### PR TITLE
Remove Turnip 

### PR DESCRIPTION
**Pull request by jessieay**
* having turnip + capybara represents an overlap in frameworks
* seem to have many more capybara specs, so should favor that
* use cases covered by turnip appear to be covered by other specs
* this will speed up our test suite - yay!

### CODE REVIEW

**Comment by phirefly** [comment]
Loving the idea of keeping one framework, but I think these specs need to move over. I'm not seeing the duplicate specs on the capybara side.

File: spec/acceptance/token_expiration.feature [hunk 1]
```diff
@@ -1,14 +0,0 @@
-Feature: Approving a proposal from the web application expires the token
-  Scenario: An approver visits the page to approve
```
Comment (jessieay) [inline spec/acceptance/token_expiration.feature]: this scenario is covered by https://github.com/18F/C2/blob/master/spec/requests/proposals_spec.rb#L85

File: spec/acceptance/login_navigation.feature [hunk 2]
```diff
@@ -1,19 +0,0 @@
...
-
-  Scenario: Logged in users
-    When I go to '/'
-    When the user is 'liono1@some-cartoon-show.com'
-    And I login
-    Then I should see alert text 'You successfully signed in'
-    And I should see 'liono1@some-cartoon-show.com'
-    And I should see 'Logout'
-
-  Scenario: Logging out
```
Comment (jessieay) [inline spec/acceptance/login_navigation.feature]: this scenario is covered by https://github.com/18F/C2/blob/master/spec/features/logout_spec.rb

File: spec/acceptance/approval_from_web_app.feature [hunk 3]
```diff
@@ -1,106 +0,0 @@
...
-    Given a proposal
-    And the proposal has an approval for 'supervisor1@test.gov' in position 1
-    Given the logged in user is 'supervisor1@test.gov'
-
-    When I go to the approval_response page without a token
-    And I should see 'Approve'
-    When I click 'Approve'
-    Then I should see alert text 'You have approved'
-
-  Scenario: An approver visits the page after previously responding
```
Comment (jessieay) [inline spec/acceptance/approval_from_web_app.feature]: this is covering the same use case as the token feature spec from the other spec file

File: spec/acceptance/approval_from_web_app.feature [hunk 4]
```diff
@@ -1,106 +0,0 @@
...
-    Given a proposal
-    And the proposal has an approval for 'supervisor1@test.gov' in position 1
-    Given the logged in user is 'supervisor1@test.gov'
-    And the proposal has been approved by the logged in user
-    When I go to the approval_response page without a token
-    And I should not see 'Approve'
-
-  Scenario: A non-approver visits the page
```
Comment (jessieay) [inline spec/acceptance/approval_from_web_app.feature]: this is covered by https://github.com/18F/C2/blob/master/spec/requests/proposals_spec.rb#L33

File: spec/acceptance/approval_from_web_app.feature [hunk 5]
```diff
@@ -1,106 +0,0 @@
...
-    Given a proposal
-    And the proposal has an approval for 'supervisor1@test.gov' in position 1
-    Given the logged in user is 'invalid-approver@test.gov'
-    When I go to the approval_response page without a token
-    Then I should not see 'Approve'
-
-  Scenario: An approver visits the page to approve in turn
```
Comment (jessieay) [inline spec/acceptance/approval_from_web_app.feature]: this is covered by https://github.com/18F/C2/blob/master/spec/requests/proposals_spec.rb#L54

File: spec/acceptance/approval_from_web_app.feature [hunk 6]
```diff
@@ -1,106 +0,0 @@
...
-    And the proposal has an approval for 'supervisor1@test.gov' in position 1
-    And the proposal has an approval for 'supervisor2@test.gov' in position 2
-    And the proposal has been approved by 'supervisor1@test.gov'
-    And the logged in user is 'supervisor2@test.gov'
-    When I go to the approval_response page without a token
-    Then I should see 'Approve'
-    When I click 'Approve'
-    Then I should see alert text 'You have approved'
-
-  Scenario: An approver visits the page to approve out of turn
```
Comment (jessieay) [inline spec/acceptance/approval_from_web_app.feature]: this is covered by https://github.com/18F/C2/blob/master/spec/features/display_status_spec.rb#L36

File: spec/acceptance/approval_from_web_app.feature [hunk 7]
```diff
@@ -1,106 +0,0 @@
...
-
-    When the logged in user is 'supervisor1@test.gov'
-    And I go to the approval_response page without a token
-    And I should see 'Approve'
-
-    When I click 'Approve'
-    Then I should see alert text 'You have approved'
-    And I should not see 'Approve'
-
-  Scenario: Displaying approval actions for a linear proposal
```
Comment (jessieay) [inline spec/acceptance/approval_from_web_app.feature]: this is covered by https://github.com/18F/C2/blob/master/spec/features/display_status_spec.rb#L36

File: spec/acceptance/approval_from_web_app.feature [hunk 8]
```diff
@@ -1,106 +0,0 @@
...
-    When the logged in user is 'supervisor1@test.gov'
-    And I go to the approval_response page without a token
-    Then I should see 'Approve'
-    When the logged in user is 'supervisor2@test.gov'
-    And I go to the approval_response page without a token
-    Then I should see 'Approve'
-    When I click 'Approve'
-    Then I should see alert text 'You have approved'
-
-  Scenario: Displaying approval actions for a parallel proposal
```
Comment (jessieay) [inline spec/acceptance/approval_from_web_app.feature]: this is covered by https://github.com/18F/C2/blob/master/spec/features/display_status_spec.rb#L2

File: spec/acceptance/login_navigation.feature [hunk 9]
```diff
@@ -1,19 +0,0 @@
-Feature: login navigation
-  Scenario: Guests
```
Comment (jessieay) [inline spec/acceptance/login_navigation.feature]: will write capybara spec for this

**Comment by jessieay** [comment]
@phirefly wrote some capybara specs to cover some of the lost coverage in 5ba04cff5a522081448149fb3b38055e7045af67 - let me know what you think!

**Comment by jessieay** [comment]
From Slack:

>  Pulling the discussion over from https://github.com/18F/C2/pull/679/files#r41395111 since others may want to chime in... what are your thoughts about pulling over the specs that are looking for text on the page e.g. "I should/should not see alert messages"? Some of this is missing in the capybara specs. What I like about the current turnip specs is that they're easy to read and pull out biz logic and they cover more of the view level specs that aren't currently in the capy specs.

> Curious if anyone else has thoughts... (edited)

> Raphy Villas [6:43 AM]
Although I like the readability that turnip provides, I don't know if it's necessary for the team. There have been questions recently about bizdev rules, which may be better addressed if specs were more readable. Case in point, the question from @jessie and @pkarman  about how many emails need to be sent out in a particular instance.

> Raphy Villas [6:47 AM]
To summarize, two questions:
1. Are we ok with capy specs being full integration
2.Does the readability of turnip provide any value at this point?

**Comment by jessieay** [comment]
@phirefly as far as I can tell, nobody on this project is actively writing Turnip specs. The directory where they live was last touched 3 months ago and when you [look back](https://github.com/18F/C2/commits/master/spec/acceptance), there haven't been new turnip specs added since February (all changes since then have been edits to existing specs because of changed behavior in the app)

Because we aren't actively using Turnip, and we have few specs written with it anyway, my recommendation is to remove it. I think the main benefit is that then we can make sure we have coverage for everything in Capybara rather than going back and forth between Capybara and Turnip to see if there are duplicates (which there already are).

To answer your questions:

1) I've used capybara for full integration many times, and we already kind of are using it here for that purpose because we have very few turnip specs anyway - so yes, very OK with it!

2) The readability is nice, but maintaining step definitions is not my favorite :)

**Comment by pkarman** [comment]
grok time for developers is lessened if we have fewer frameworks/tools in play. If the missing piece is improved documentation (comments) then +1 for more comments! (looking at you @jessieay)

**Comment by jessieay** [comment]
@pkarman MORE COMMENTS? IN OUR CODE? :crying_cat_face:

**Comment by pkarman** [comment]
I'm +1 to drop turnip if we can compensate by commenting any remaining tests that require biz rule clarification. E.g. "how many emails should we see?" might require a comment or two to fully explain.

**Comment by jessieay** [comment]
@pkarman I bet we could fit that in spec descriptions of we use nested scenarios!

**Comment by pkarman** [comment]
I'm all for the shoehorn until the shoes just don't fit.

**Comment by jessieay** [comment]
:athletic_shoe: :postal_horn:

**Comment by jessieay** [comment]
This is ready for re-review / merge

**Comment by pkarman** [comment]
I'm +1 to merge this, but would like @phirefly to weigh in again since he has more skin in the Turnip game.

**Comment by afeld** [comment]
I'm not a fan of Gherkin either, and while I understand the "this could be read by stakeholders" argument, that doesn't actually happen ever (well, I've heard of it happening a little kinda like once in another company). :100: :+1: to merging. Will let @phirefly click the button.

**Comment by phirefly** [comment]
:heart: the communication. Thanks all!
:shipit:

**Pull request merged**